trieste/acquisition/function.py

```diff
@@ -130,6 +133,117 @@ def expected_improvement(model: ModelInterface, eta: tf.Tensor, at: QueryPoints)
     return (eta - mean) * normal.cdf(eta) + variance * normal.prob(eta)
 
 
+class MaxValueEntropySearch(SingleModelAcquisitionBuilder):
+    """
+    Builder for the max-value entropy search acqusiiton function (for function minimisation)
+    """
+
+    def __init__(self,search_space: SearchSpace, num_samples: int = 10, grid_size: int = 5000):
+        """
+        :param num_samples: integer determining how many samples to draw of the minimum 
+            (does not need to be large).
+        :param grid_size: number of random locations in grid used to fit the gumbel distribution 
+            and approximately generate the samples of the minimum (recommend scaling with problem dimension).
+        """
+        self._num_samples = num_samples
+        self._grid_size = grid_size
+        self._search_space = search_space
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ModelInterface
+    ) -> AcquisitionFunction:
+        """
+        Need to sample possible min-values from our posterior. To do this we implement a Gumbel sampler.
+        we approximate Pr(y*^hat<y) by Gumbel(alpha,beta) then sample from Gumbel.
+
+        :param dataset: Unused.
+        :param model: The model over the specified ``dataset``.
+        :return: The MES function.
+        """
+
+        # generate random grid to fit Gumbel distribution
+        # grid consists of already queried locations and grid_size other random locations
+        query_points = self._search_space.sample(self._grid_size)
+        query_points = tf.concat([dataset.query_points,query_points],0)
+        fmean, fvar = model.predict(query_points)
+        fsd = tf.math.sqrt(fvar)
+        idx = tf.math.argmax(-fmean[:tf.shape(dataset.query_points)[0]])
+
+        # Fit Gumbel distriubtion
+        # scaling so that gumbel scale is proportional to IQ range of cdf Pr(y*<z)
+        # find quantiles Pr(y*<y1)=r1 and Pr(y*<y2)=r2
+        right = tf.squeeze(fmean[tf.squeeze(idx)])
+        left = right
```

Comment:
this could be `left = right = tf.squeeze(fmean[tf.squeeze(idx)])` since `tf.Tensor`s are immutable, but it's up to you

Reply:
alternatively, since we recalculate the posterior best later,
```suggestion
        posterior_best = tf.squeeze(fmean[tf.squeeze(idx)])
        left = right = posterior_best
```
then reuse `posterior_best` later

Reply:
I have changed this bit now (following the MES implementation in BOTorch).

My approach here turns out to be overkill.

Reply:
was the first approach one of your own?